### PR TITLE
Add minesweeper exercise

### PR DESCRIPTION
**Pull request by glennj**

### CODE REVIEW

File: exercises/practice/minesweeper/.meta/example.sh [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+# some global vars
+declare -A board
+height=$#
+width=${#1}
+
+parse_input() {
+    local rownum=0 row col index char
+    for row in "$@"; do
+        for ((col=0; col < "${#row}"; col++)); do
```
Comment (IsaacG) [inline exercises/practice/minesweeper/.meta/example.sh:11]: Is there a reason you space pad the `<` operator but not the `=` operator?

**Review by IsaacG — APPROVED** [review]

File: exercises/practice/minesweeper/.meta/example.sh (hunk 1 above)
Comment (kotp) [reply exercises/practice/minesweeper/.meta/example.sh:11]: I, of course, cannot answer this question for the author, but it does seem to be consistent with the other assignments such as line 5 and 6.  I am all for space around operators where allowed by the language though, as it makes the "words", well, words.

File: exercises/practice/minesweeper/minesweeper.bats [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+}
+
+@test "mine surrounded by spaces" {
+    [[ $BATS_RUN_SKIPPED == "true" ]] || skip
+    input=(
+        "   "
+        " * "
+        "   "
+    )
+    run bash minesweeper.sh "${input[@]}"
```
Comment (glennj) [inline exercises/practice/minesweeper/minesweeper.bats:59]: My main concern was how to represent an array of strings, particularly whitespace-only strings, as input to the script. This method OK with everyone @here?

Comment (IsaacG) [reply exercises/practice/minesweeper/minesweeper.bats:59]: It gets the job done so no objections here. You can also use STDIN and `printf "%s\n" "${data[@]}" | ...`.

**Review by IsaacG — APPROVED** [review]

File: exercises/practice/minesweeper/minesweeper.bats (hunk 2 above)
Comment (glennj) [reply exercises/practice/minesweeper/minesweeper.bats:59]: I did try that, thought it was too messy.

**Pull request merged**